internal/controllers/nft_controller.go

```diff
@@ -837,3 +840,200 @@ func (nc *NFTController) GetVinCredential(c *fiber.Ctx) error {
+	}
+	tid := types.NewNullDecimal(new(decimal.Big).SetBigMantScale(ti, 0))
+
+	client := registry.Client{
+		Producer:     udc.producer,
+		RequestTopic: "topic.transaction.request.send",
+		Contract: registry.Contract{
+			ChainID: big.NewInt(udc.Settings.DIMORegistryChainID),
+			Address: common.HexToAddress(udc.Settings.DIMORegistryAddr),
+			Name:    "DIMO",
+			Version: "1",
+		},
+	}
+
+	tx, err := udc.DBS().Reader.BeginTx(c.Context(), nil)
+	if err != nil {
+		return fmt.Errorf("could not create transaction: %w", err)
+	}
+	defer tx.Rollback() //nolint
+
+	vehicleNFT, err := models.VehicleNFTS(
+		models.VehicleNFTWhere.TokenID.EQ(tid),
+		qm.Load(qm.Rels(models.VehicleNFTRels.UserDevice)),
+		qm.Load(qm.Rels(models.VehicleNFTRels.MintRequest)),
+		qm.Load(qm.Rels(models.VehicleNFTRels.VehicleTokenAftermarketDevice)),
+		qm.Load(qm.Rels(models.VehicleNFTRels.VehicleTokenSyntheticDevice)),
+		qm.Load(qm.Rels(models.VehicleNFTRels.UserDevice, models.UserDeviceRels.UserDeviceAPIIntegrations)),
+	).One(c.Context(), tx)
+	if err != nil {
+		return fiber.NewError(fiber.StatusBadRequest, "failed to find device by token id: %s", tis)
+	}
+
+	if vehicleNFT.R.UserDevice == nil {
+		udc.log.Warn().Msgf("no user device associated with NFT %s", tis) // this should never happen
+		return fiber.NewError(fiber.StatusBadRequest, "no user device associated with NFT")
+	}
+
+	bvs, user, err := udc.checkDeviceBurn(c.Context(), vehicleNFT, vehicleNFT.R.UserDevice.UserID)
+	if err != nil {
+		return fiber.NewError(fiber.StatusBadRequest, err.Error())
+	}
+
+	br := new(BurnRequest)
+	if err := c.BodyParser(br); err != nil {
+		return fiber.NewError(fiber.StatusBadRequest, "Couldn't parse request body.")
+	}
+
+	udc.log.Info().
+		Interface("httpRequestBody", br).
+		Interface("client", client).
+		Interface("burnVehicleSign", bvs).
+		Interface("typedData", client.GetPayload(&bvs)).
+		Msg("Got request.")
+
+	hash, err := client.Hash(&bvs)
+	if err != nil {
+		return fmt.Errorf("could not hash bvs: %w", err)
+	}
+
+	sigBytes := common.FromHex(br.Signature)
+	recAddr, err := helpers.Ecrecover(hash, sigBytes)
+	if err != nil {
+		return err
+	}
+
+	realAddr := common.HexToAddress(*user.EthereumAddress)
+	if recAddr != realAddr {
+		return fiber.NewError(fiber.StatusBadRequest, "signature incorrect")
+	}
+
+	requestID := ksuid.New().String()
+	mtr := models.MetaTransactionRequest{
+		ID:     requestID,
+		Status: "Unsubmitted",
+	}
+
+	if err := mtr.Insert(c.Context(), tx, boil.Infer()); err != nil {
+		return err
+	}
+
+	vehicleNFT.BurnRequestID = null.StringFrom(requestID)
+	if _, err := vehicleNFT.Update(c.Context(), tx, boil.Infer()); err != nil {
+		return fiber.NewError(fiber.StatusBadRequest, "failed to associate burn request with vehicle")
+	}
+
+	if err := tx.Commit(); err != nil {
+		return err
+	}
+
+	udc.log.Info().Msgf("submitted metatransaction request %s", requestID)
```

Comment:
nit:
Maybe drop this down to debug since we do not need to know every time this function is run.